app/helpers/admin/dashboard_helper.rb

```diff
@@ -1,10 +1,41 @@
 # frozen_string_literal: true
 
 module Admin::DashboardHelper
-  def feature_hint(feature, enabled)
-    indicator   = safe_join([enabled ? t('simple_form.yes') : t('simple_form.no'), fa_icon('power-off fw')], ' ')
-    class_names = enabled ? 'pull-right positive-hint' : 'pull-right neutral-hint'
+  def relevant_account_ip(account, ip_query)
+    default_ip = [account.user_current_sign_in_ip || account.user_sign_up_ip]
 
-    safe_join([feature, content_tag(:span, indicator, class: class_names)])
+    matched_ip = begin
+      ip_query_addr = IPAddr.new(ip_query)
+      account.user.recent_ips.find { |(_, ip)| ip_query_addr.include?(ip) } || default_ip
+    rescue IPAddr::Error
+      default_ip
+    end.last
+
+    if matched_ip
+      link_to matched_ip, admin_accounts_path(ip: matched_ip)
+    else
+      '-'
+    end
+  end
+
+  def relevant_account_timestamp(account)
+    timestamp, exact = begin
+      if account.user_current_sign_in_at && account.user_current_sign_in_at < 24.hours.ago
+        [account.user_current_sign_in_at, true]
+      elsif account.user_current_sign_in_at
+        [account.user_current_sign_in_at, false]
```

Comment:
Might be slightly confusing if it shows “Today” for a log in that was actually 23h ago. Not sure how it's handled elsewhere.

Reply:
The reason I did it this way is we only bump current_sign_in_at once every 24 hours. So if it's less than 24h ago, it may give a misleading sense of precision.